docs/rules/no-func-assign.md

```diff
@@ -48,4 +48,8 @@ function foo(foo) { // `foo` is shadowed.
 function foo() {
     var foo = bar;  // `foo` is shadowed.
 }
+
+var a = function hello() {
+  hello = 123;
+};
```

Comment:
This should be incorrect code for this rule, as in [this demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tZnVuYy1hc3NpZ246IFwiZXJyb3JcIiovXG5cbnZhciBhID0gZnVuY3Rpb24gaGVsbG8oKSB7XG4gIGhlbGxvID0gMTIzO1xufTtcblxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19). We've agreed to document the actual behavior of this rule regarding named function expressions.

Reply:
Updated 👍🏻 , Please take a look.